setup.py

```diff
@@ -1,14 +1,67 @@
-from setuptools import setup, find_packages
+from setuptools import setup, find_packages, Extension
+from glob import glob
+try:
+    from Cython.Build import cythonize
+    from Cython.Compiler.Options import get_directive_defaults
+
+    use_cython = True
+except ImportError:
+    use_cython = False
+import numpy as np
+import os
+import sys
+
+define_macros = [("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION")]
+
+if use_cython:
+    suffix = ".pyx"
+    directive_defaults = get_directive_defaults()
+    directive_defaults["binding"] = True
+    directive_defaults["language_level"] = 3
+    if os.environ.get("CYTHON_COVERAGE"):
+        directive_defaults["linetrace"] = True
+        define_macros.append(("CYTHON_TRACE_NOGIL", "1"))
+else:
+    suffix = ".c"
+
+include_dirs = [np.get_include(), os.path.join(sys.prefix, "include")]
+ext_modules = [
+    Extension(
+        name[: -len(suffix)].replace("/", ".").replace("\\", "."),
+        [name],
+        include_dirs=include_dirs,
+        define_macros=define_macros,
+    )
+    for name in glob(f"suitesparse_graphblas/**/*{suffix}", recursive=True)
+]
+if use_cython:
+    ext_modules = cythonize(ext_modules, include_path=include_dirs)
+
+with open("README.md") as f:
+    long_description = f.read()
+
+package_data = {"suitesparse_graphblas": ["*.pyx", "*.pxd", "*.h"]}
+if sys.platform == "win32":
+    package_data["suitesparse_graphblas"].append("*.dll")
 
 setup(
     name='suitesparse-graphblas',
-    version='4.0.3',
+    version=versioneer.get_version(),
+    cmdclass=versioneer.get_cmdclass(),
     description='SuiteSparse:GraphBLAS Python bindings.',
+    long_description=long_description,
+    long_description_content_type="text/markdown",
     packages=find_packages(),
     author='Michel Pelletier, James Kitchen, Erik Welch',
+    author_email="jim22k@gmail.com,erik.n.welch@gmail.com",
```

Comment:
Go ahead and use michel@graphegon.com for my author email as we discussed.

Reply:
Done.